compiler/rustc_query_system/src/dep_graph/serialized.rs

```diff
@@ -138,9 +153,35 @@ impl SerializedDepGraph {
         self.nodes[dep_node_index]
     }
 
+    /// This computes and sets up the index for just the specified `DepKind`.
+    fn setup_index(&self, dep_kind: DepKind) {
+        let _timer = self.prof.generic_activity("incr_comp_dep_graph_setup_index");
+
+        let mut index = UnhashMap::with_capacity_and_hasher(
+            self.index_sizes[dep_kind.as_usize()],
+            Default::default(),
+        );
+
+        for (idx, node) in self.nodes.iter_enumerated() {
+            if node.kind == dep_kind {
+                index.insert(node.hash, idx);
+            }
+        }
+
+        // This may race with the prefetching thread, but that will set the same value.
+        self.index[dep_kind.as_usize()].set(index).ok();
```

Comment:
Is same-value race ok in rust?

Reply:
It's not a data race as that would indeed be unsafe.